app/src/org/commcare/views/media/MediaLayout.java

```diff
@@ -541,49 +416,39 @@ private boolean useResizingImageView() {
                         || "width".equals(ResizingImageView.resizeMethod));
     }
 
-    /**
-     * @return The appropriate max size of an image view pane in this widget. returned as an int
-     * array of [width, height]
-     */
-    private int[] getMaxCenterViewBounds() {
-        DisplayMetrics metrics = this.getContext().getResources().getDisplayMetrics();
-        int maxWidth = metrics.widthPixels;
-        int maxHeight = metrics.heightPixels;
-
-        // subtract height for textview and buttons, if present
-        if (viewText != null) {
-            maxHeight = maxHeight - viewText.getHeight();
-        }
-        if (videoButton != null) {
-            maxHeight = maxHeight - videoButton.getHeight();
-        } else if (audioButton != null) {
-            maxHeight = maxHeight - audioButton.getHeight();
-        }
-
-        // reduce by third for safety
-        return new int[]{maxWidth, (2 * maxHeight) / 3};
+    private void initView(Context context) {
+        View view = LayoutInflater.from(context).inflate(R.layout.media_layout, this);
+        audioButton = view.findViewById(R.id.audio_button);
+        videoButton = view.findViewById(R.id.video_button);
+        textViewContainer = view.findViewById(R.id.question_text_container);
+        videoView = view.findViewById(R.id.inline_video_view);
+        qrView = view.findViewById(R.id.qr_view);
+        imageView = view.findViewById(R.id.image);
+        resizingImageView = view.findViewById(R.id.resizing_image);
+        downloadIcon = view.findViewById(R.id.download_media_icon);
+        progressBar = view.findViewById(R.id.progress_bar);
+        missingMediaText = view.findViewById(R.id.missing_media_tv);
+        divider = view.findViewById(R.id.divider);
+        mediaLayoutBottomBarrier = view.findViewById(R.id.media_barrier);
+        missingMediaBackground = view.findViewById(R.id.missing_media_background);
+
+        resetView();
     }
 
-    /**
-     * This adds a divider at the bottom of this layout. Used to separate
-     * fields in lists.
-     */
-    public void addDivider(ImageView v) {
-        RelativeLayout.LayoutParams dividerParams =
-                new RelativeLayout.LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT);
-        if (missingMediaView != null) {
-            dividerParams.addRule(RelativeLayout.BELOW, missingMediaView.getId());
-        } else if (videoButton != null) {
-            dividerParams.addRule(RelativeLayout.BELOW, videoButton.getId());
-        } else if (audioButton != null) {
-            dividerParams.addRule(RelativeLayout.BELOW, audioButton.getId());
-        } else if (viewText != null) {
-            // No picture
-            dividerParams.addRule(RelativeLayout.BELOW, viewText.getId());
-        } else {
-            Log.e(TAG, "Tried to add divider to uninitialized ATVWidget");
-            return;
-        }
-        addView(v, dividerParams);
+    private void resetView() {
```

Comment:
this method can be avoided by setting the initial visibility of all views as gone in xml layout itself.